src/Database/LSMTree/Normal.hs

```diff
@@ -275,6 +293,36 @@ retrieveBlobs = undefined
   Snapshots
 -------------------------------------------------------------------------------}
 
+{-# SPECIALISE snapshot' :: TableHandle IO k v blob -> IO (V.Vector (V.Vector Internal.RunFsPaths)) #-}
+-- | Report the names of the runs that are in use by a table. This can be used
+-- to (re-)open a table from disk using 'open''.
+--
+-- NOTE: this persists the write buffer to disk as part of the snapshot. The
+-- write buffer in the table is not flushed.
+--
+-- TODO: this is a temporary solution until we have snapshots, which will be the
+-- /only/ method of persisting the state of the table.
+snapshot' :: IOLike m => TableHandle m k v blob -> IO (V.Vector (V.Vector Internal.RunFsPaths))
+snapshot'  (TableHandle th) = Internal.snapshot' th
+
+{-# SPECIALISE open' :: Session IO -> Internal.TableConfig ->  V.Vector (V.Vector Internal.RunFsPaths) -> IO (TableHandle IO k v blob) #-}
+-- | Open a table from disk given a set of runs to load.
+--
+-- TODO: this is a temporary solution until we have snapshots, which will be the
+-- /only/ method of persisting the state of the table.
+open' ::
+     IOLike m
+  => Session m
+  -> Internal.TableConfig
+     -- ^ The table config is not persisted to disk (yet) until we have
+     -- snapshots. The user has to make sure to only load runs that were created
+     -- with the same configuration, or the behaviour might be undefined.
+  -> V.Vector (V.Vector Internal.RunFsPaths)
+     -- ^ Runs to open, identified by their 'RunFsPaths's. Acquire this
+     -- structure using 'snapshot''.
+  -> m (TableHandle m k v blob)
+open' (Session sesh) conf paths = TableHandle <$> Internal.open' sesh conf paths
```

Comment:
An alternative I've been thinking of: write the `RunFsPaths`s to disk instead of passing them in memory, and use the `SnapshotName` to identify the file that contains the `RunFsPaths`s